src/common/types/vector.cpp

```diff
@@ -272,7 +272,14 @@ Value Vector::GetValue(idx_t index) const {
 		return Value::DOUBLE(((double *)data)[index]);
 	case TypeId::VARCHAR: {
 		auto str = ((string_t *)data)[index];
-		return Value::BLOB(str.GetString());
+		auto data_str = str.GetData();
+		// avoiding implicit cast and double conversion
+		if(str.GetSize() >= 2 && data_str[0] == '\\' && data_str[1] == 'x') {
+			return Value::BLOB(str.GetString(), false);
```

Comment:
You can always pass false to must_cast here, no? This check here seems unnecessary.

Reply:
I think you're right, because at this point, it makes sense to avoid any implicit cast.